pandas/tests/frame/methods/test_rank.py

```diff
@@ -329,3 +331,88 @@ def test_pct_max_many_rows(self):
         )
         result = df.rank(pct=True).max()
         assert (result == 1).all()
+
+    @pytest.mark.parametrize(
+        "contents,dtype",
+        [
+            (
+                [
+                    -np.inf,
+                    -50,
+                    -1,
+                    -1e-20,
+                    -1e-25,
+                    -1e-50,
+                    0,
+                    1e-40,
+                    1e-20,
+                    1e-10,
+                    2,
+                    40,
+                    np.inf,
+                ],
+                "float64",
+            ),
+            (
+                [
+                    -np.inf,
+                    -50,
+                    -1,
+                    -1e-20,
+                    -1e-25,
+                    -1e-45,
+                    0,
+                    1e-40,
+                    1e-20,
+                    1e-10,
+                    2,
+                    40,
+                    np.inf,
+                ],
+                "float32",
+            ),
+            ([np.iinfo(np.uint8).min, 1, 2, 100, np.iinfo(np.uint8).max], "uint8"),
+            pytest.param(
+                [
+                    np.iinfo(np.int64).min,
+                    -100,
+                    0,
+                    1,
+                    9999,
+                    100000,
+                    1e10,
+                    np.iinfo(np.int64).max,
+                ],
+                "int64",
+                marks=pytest.mark.xfail(
+                    reason="iNaT is equivalent to minimum value of dtype"
+                    "int64 pending issue GH#16674"
+                ),
+            ),
+            ([NegInfinity(), "1", "A", "BA", "Ba", "C", Infinity()], "object"),
+        ],
+    )
+    def test_rank_inf_and_nan(self, contents, dtype):
```

Comment:
Mostly duplicated from https://github.com/pandas-dev/pandas/blob/0805043ffab989f29444711f4be2c4cc26325494/pandas/tests/series/methods/test_rank.py#L269
Could deduplicate some of this into a fixture in a followup if that makes sense

Reply:
I think that would make sense - besides, the `black` formatter doesn't make this look great